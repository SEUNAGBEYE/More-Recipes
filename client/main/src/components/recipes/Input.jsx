import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description - Input Component
 *
 * @param {Object} props
 *
 * @returns {Jsx} Jsx
 */
const Input = (props) => (
  <div>
    <fieldset>
      <label> {props.name} { props.number}</label>
      <input className="form-control"
        onBlur={props.onChange}
        name={props.name}
        id={props.id}
        data-index={props.index}
        data-name={`${props.name}-${props.index}`}
        defaultValue={props.value}
        required/>
      <span className="icons fa fa-trash" id={props.id}
        data-index={props.index}
        data-name={`${props.name}-${props.index}`}
        onClick={props.removeInput}
      />
    </fieldset>
  </div>
);

const propTypes = {
  name: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  removeInput: PropTypes.func.isRequired,
  number: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
};

Input.propTypes = propTypes;

export default Input;
