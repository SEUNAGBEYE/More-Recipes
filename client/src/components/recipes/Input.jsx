import React from 'react';

export default (props) => (
  <div>
    <fieldset>
      <label> {props.name} { props.number}</label>
      <input className="steps form-control"
        onBlur={props.onChange}
        name={props.name}
        id={props.id}
        defaultValue={props.value}
        required/>
    </fieldset>
  </div>
);
