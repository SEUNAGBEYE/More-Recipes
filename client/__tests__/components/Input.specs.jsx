
import React from 'react';
import { shallow } from 'enzyme';
import Input from '../../main/src/components/recipes/Input';


const recipe = {
  name: 'Amala',
  image: 'This is the image'
}

const props = {
  name: 'Input One',
  value: 'Value',
  onChange: jest.fn(),
  number: 1,
  id: 1
};

describe('# CategoryButton', () => {
  const wrapper = shallow(
      <Input {...props} />
  );
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot();
    done();
  });

});
