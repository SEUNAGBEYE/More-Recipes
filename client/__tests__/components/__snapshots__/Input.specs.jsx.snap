// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# CategoryButton should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    id={1}
    name="Input One"
    number={1}
    onChange={[MockFunction]}
    value="Value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <fieldset>
        <label>
           
          Input One
           
          1
        </label>
        <input
          className="form-control"
          data-index={undefined}
          data-name="Input One-undefined"
          defaultValue="Value"
          id={1}
          name="Input One"
          onBlur={[MockFunction]}
          required={true}
        />
        <span
          className="icons fa fa-trash"
          data-index={undefined}
          data-name="Input One-undefined"
          id={1}
          onClick={undefined}
        />
      </fieldset>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
             
            Input One
             
            1
          </label>,
          <input
            className="form-control"
            data-index={undefined}
            data-name="Input One-undefined"
            defaultValue="Value"
            id={1}
            name="Input One"
            onBlur={[MockFunction]}
            required={true}
          />,
          <span
            className="icons fa fa-trash"
            data-index={undefined}
            data-name="Input One-undefined"
            id={1}
            onClick={undefined}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Input One",
              " ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Input One",
            " ",
            1,
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "data-index": undefined,
            "data-name": "Input One-undefined",
            "defaultValue": "Value",
            "id": 1,
            "name": "Input One",
            "onBlur": [MockFunction],
            "required": true,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icons fa fa-trash",
            "data-index": undefined,
            "data-name": "Input One-undefined",
            "id": 1,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "fieldset",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <fieldset>
          <label>
             
            Input One
             
            1
          </label>
          <input
            className="form-control"
            data-index={undefined}
            data-name="Input One-undefined"
            defaultValue="Value"
            id={1}
            name="Input One"
            onBlur={[MockFunction]}
            required={true}
          />
          <span
            className="icons fa fa-trash"
            data-index={undefined}
            data-name="Input One-undefined"
            id={1}
            onClick={undefined}
          />
        </fieldset>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
               
              Input One
               
              1
            </label>,
            <input
              className="form-control"
              data-index={undefined}
              data-name="Input One-undefined"
              defaultValue="Value"
              id={1}
              name="Input One"
              onBlur={[MockFunction]}
              required={true}
            />,
            <span
              className="icons fa fa-trash"
              data-index={undefined}
              data-name="Input One-undefined"
              id={1}
              onClick={undefined}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Input One",
                " ",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Input One",
              " ",
              1,
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "data-index": undefined,
              "data-name": "Input One-undefined",
              "defaultValue": "Value",
              "id": 1,
              "name": "Input One",
              "onBlur": [MockFunction],
              "required": true,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icons fa fa-trash",
              "data-index": undefined,
              "data-name": "Input One-undefined",
              "id": 1,
              "onClick": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "fieldset",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
